File: src/app/api/models/Component.ts
import mongoose from 'mongoose';

export const COMPONENT_STATUS = ['operational', 'degraded', 'outage', 'maintenance'] as const;
export type ComponentStatus = typeof COMPONENT_STATUS[number];

const componentSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  description: String,
  status: {
    type: String,
    enum: COMPONENT_STATUS,
    default: 'operational',
  },
  lastChecked: Date,
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  }
});

// Update the updatedAt timestamp before saving
componentSchema.pre('save', function(next) {
  this.updatedAt = new Date();
  next();
});

export const Component = mongoose.models.Component || mongoose.model('Component', componentSchema); 

File: web/lib/models/component.dart
class Component {
  final String id;
  final String name;
  final String description;
  final String status; // 'operational', 'degraded', 'partial', 'major', 'under_maintenance'
  final String createdAt;
  final String updatedAt;

  Component({
    required this.id,
    required this.name,
    required this.description,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });
}
